secret-sync-controller/main.go

```diff
@@ -0,0 +1,157 @@
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8085", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&providerVolumePath, "provider-volume", "/provider", "Volume path for provider.")
+	flag.IntVar(&maxCallRecvMsgSize, "max-call-recv-msg-size", 1024*1024*4, "maximum size in bytes of gRPC response from plugins")
+	flag.StringVar(&tokenRequestAudience, "token-request-audience", "", "Audience for the token request.")
+
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme: scheme,
+		Metrics: server.Options{
+			BindAddress: metricsAddr,
+		},
+		WebhookServer:          webhook.NewServer(webhook.Options{Port: 9443}),
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "29f1d54e.secret-sync.x-k8s.io",
+		// LeaderElectionReleaseOnCancel defines if the leader should step down voluntarily
+		// when the Manager ends. This requires the binary to immediately end when the
+		// Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
+		// speeds up voluntary leader transitions as the new leader don't have to wait
+		// LeaseDuration time first.
+		//
+		// In the default scaffold provided, the program ends immediately after
+		// the manager stops, so would be fine to enable this option. However,
+		// if you are doing or is intended to do any operation such as perform cleanups
+		// after the manager stops then its usage might be unsafe.
+		// LeaderElectionReleaseOnCancel: true,
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// token request client
+	kubeClient := kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	tokenClient := k8s.NewTokenClient(kubeClient)
+	if err != nil {
+		setupLog.Error(err, "failed to create token client")
+		os.Exit(1)
+	}
+
+	providerClients := provider.NewPluginClientBuilder([]string{providerVolumePath}, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxCallRecvMsgSize)))
+	defer providerClients.Cleanup()
+
+	if err = (&controllers.SecretSyncReconciler{
+		Clientset:       kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie()),
+		Client:          mgr.GetClient(),
+		Scheme:          mgr.GetScheme(),
+		TokenClient:     tokenClient,
+		ProviderClients: providerClients,
+		TokenRequest: &tokenutil.TokenRequest{
+			Audience: tokenRequestAudience,
+		},
```

Comment:
The reconciler/token generator shouldn't be performing the split based on the delimiter, all that should be done here. The field in reconciler struct should be `Audiences []string` instead of `TokenRequest`. The token request audience should be split on the delimiter, perform any validaiton (if any) and set the list of audiences in the reconciler struct.